common/src/pcl_base.cpp

```diff
@@ -70,7 +70,7 @@ pcl::PCLBase<pcl::PCLPointCloud2>::setInputCloud (const PCLPointCloud2ConstPtr &
 
   // Obtain the size of all fields. Restrict to sizeof FLOAT32 for now
   field_sizes_.resize (input_->fields.size ());
-  for (std::size_t d = 0; d < input_->fields.size (); ++d)
+  for (int d = 0; d < static_cast<int>(field_sizes_.size ()); ++d)
```

Comment:
Can you explain this change?

Reply:
The assumption is that `fields.size() < INT_MAX`, so there's no need to make that loop using `std::size_t`. We can codify it. `using PointSizeT = std::size_t; using FieldSizeT = int`

Reply:
Well true, I think it's common sense to assume we have at most couple dozen fields in a point cloud. But is `fields.size() < INT_MAX` explicitly stated anywhere? And even if so, I'm not sure explicit cast to `int` makes sense. It does not _enforce_ the convention, rather breaks silently if the number of fields is too large.

Reply:
AFAIK, no. But In several other places, this silent breaking on contract is done. Maybe enforce it in PCLField or related class?

Reply:
> Maybe enforce it in PCLField or related class?

Perhaps, but please elaborate on what you have in mind.

Reply:
stdlib enforces that `size()` returns `std::size_t`. So, we'll need to `static_cast`. One way to enforce the size limit would be the places where I've seen fields being added:
`POINT_CLOUD_REGISTER_POINT_STRUCT`, `FieldAdder` and `concatenateFields`.

The first one is pure-preprocessor but I don't see a way to limit the total items in a preprocessor friendly manner.

But we can enforce the `INT_MAX` or `UINT_MAX` in `conversions.h`, `io.hpp` and `io.cpp` for the other two structs/functions. Enforcing would just involve throwing an exception because that's a really extraordinary situation

Reply:
Nothing except the fact that it's not needed (now 😝 ). Maybe using uint64 makes sense because I don't think the headaches of 2 separate max_range makes sense for the measly space savings the small number of field indices will give (We'll use 50% more memory 🤷‍♂ of a small vector)

Reply:
So, should I go ahead and remove cast and assumption of int from `field_indices.size()`?

Reply:
Having had that discussion, I'd say yes.